src/validation.cpp

```diff
@@ -1906,11 +1907,15 @@ bool CChainState::ConnectBlock(const CBlock& block, CValidationState& state, CBl
         {
             CAmount txfee = 0;
             if (!Consensus::CheckTxInputs(tx, state, view, pindex->nHeight, txfee)) {
+                // CheckTxInputs may return MISSING_INPUTS but we can't return that, as
+                // its nonsense for a block, so we reset the reason flag to CONSENSUS here.
+                state.Invalid(ValidationInvalidReason::CONSENSUS, false,
```

Comment:
Make it `CONSENSUS` only if `NOT_STANDARD`? I know that atm only those 2 are set but, imo, it's more clear to do so inside that condition:
```cpp
if (state.GetReason() == ValidationInvalidReason::NOT_STANDARD) {
    state.Invalid(ValidationInvalidReason::CONSENSUS, ...);
}
```
The same happens in line 1951.

Reply:
Done.

Reply:
ATM `CheckTxInputs` set `MISSING_INPUTS` and `CONSENSUS` reasons and only makes sense to override when reason is `MISSING_INPUTS`:
```cpp
if (state.GetReason() == ValidationInvalidReason::MISSING_INPUTS) {
    state.Invalid(ValidationInvalidReason::CONSENSUS, ...);
}
```

Reply:
Errrr....fixed, actually this time.